src/cmake/externalpackages.cmake

```diff
@@ -100,19 +100,59 @@ checked_find_package (pugixml 1.8 REQUIRED)
 
 
 # LLVM library setup
-checked_find_package (LLVM 7.0 REQUIRED
-                      PRINT LLVM_SYSTEM_LIBRARIES CLANG_LIBRARIES)
-# ensure include directory is added (in case of non-standard locations
-include_directories (BEFORE SYSTEM "${LLVM_INCLUDES}")
-link_directories ("${LLVM_LIB_DIR}")
+
+############ HACK ##############
+# On OSX, the Homebrew (and maybe any build) of LLVM 10.0 seems to have a
+# link conflict with its dependency on the llvm libc++ and the system
+# libc++, both can end up dynamically linked and lead to very subtle and
+# frustrating behavior failures (in particular, osl's use of libclang will
+# botch include file parsing any time LD_LIBRARY_PATH doesn't have the llvm
+# libc++ first).
+#
+# It seems that this is not a problem when linking against the llvm and
+# libclang libraries statically. So on apple and when LLVM 10+ are involved,
+# just force that choice. Other than larger executables, it seems harmless,
+# and in any case a better choice than this beastly bug.
+#
+# We can periodically revisit this with new version of LLVM, maybe they will
+# fix things and we won't require this preemptive static linking.
+if (APPLE AND LLVM_VERSION VERSION_GREATER_EQUAL 10.0)
+    set (LLVM_STATIC ON)
+endif ()
+
+checked_find_package(LLVM REQUIRED CONFIG
+                     HINTS ${LLVM_DIRECTORY})
+
+# manual version check because "LLVM is API-compatible only with matching major.minor versions" aka it requires VERSION_EQUAL
+if ( ${LLVM_PACKAGE_VERSION} VERSION_LESS 7 )
+	message(FATAL_ERROR "LLVM 7.0+ is required but ${LLVM_PACKAGE_VERSION} was found")
+endif()
+
+include_directories(BEFORE SYSTEM ${LLVM_INCLUDE_DIRS})
+add_definitions(${LLVM_DEFINITIONS})
+link_directories(${LLVM_LIBRARY_DIRS})
```

Comment:
Are these still necessary to be declared here and therefore be exposed to the whole of the osl code base? Or are exported targets set up correctly by the LLVM config cmake? In other words, should these be deleted, and instead liboslexec and other components should have `target_link_libaries` say `PRIVATE llvm::foo` for whatever specific components they use?

Reply:
Kinda, for the clang libraries this is trivial. For LLVM itself, it is a bit more complicated.
It seems that OSL depends on 2 target independent libs (MCJIT and Passes) and 3 target dependent (Codegen, Disassembler, AsmParser). 
For the independent once, it is trivial to target them directly. 
For the target dependent libs, it is a bit harder as they contain the target in the lib name (for example `LLVMX86CodeGen`).
At first I just took the route of adding them explicit, but later reverted to the "target all what the installed llvm version supports".
That is why I use the `llvm_map_components_to_libnames` in order to support arbitrary targets. 
So I guess the question is, does OSL support anything other than x86 and NVPTX ? 
Ff not, it is trivial to directly target those components.
If we want to support all possible targets we needs some additional setup (though we could probably limit it to oslexec). Note that I think it can be further simplified  upon bumping the minimum required version to LLVM8 as then have all the special `allXXX` targets.

Reply:
Currently, OSL doesn't support anything but x86 and NVPTX, but I expect it will need to fairly soon (at the very least, the ARM-based Macs that are coming.